climpred/classes.py

```diff
@@ -84,20 +88,119 @@ class PredictionEnsemble:
     should house functions that both ensemble types can use.
     """
 
+    # ---------------
+    # Magic Functions
```

Comment:
just for my own knowledge, are these officially named magic functions?

Reply:
@ahuang11, actually, magic methods. I'll change it. But it's the dunderscore methods that add "magic" to classes: https://rszalski.github.io/magicmethods/